### PR TITLE
Add codecov coverage uploading

### PR DESCRIPTION
**Pull request by filmaj**
###  Summary

Describe the goal of this PR. Mention any related Issue numbers.

### Requirements (place an `x` in each `[ ]`)

* [ ] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/slack-github-action/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [ ] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

**Comment by filmaj** [comment]
Hmm why didn't CI run my changes to the workflow file? 🤔 
https://github.com/slackapi/slack-github-action/actions/runs/9008262092/job/24749902585

**Comment by zimeg** [comment]
@filmaj hmm that's super strange. [The workflow file changes seem to be noticed](https://github.com/slackapi/slack-github-action/actions/runs/9008262092/workflow) but I'd guess the `pull_request_target` might use the **workflow** on `main` with the **changes** of the PR branch being checked out.

For testing purposes, I'm wondering if changing this to `pull_request` would fix this? Just to test though. I think you have the magic to use repository secrets on pull request, which motivated the **staging** environments and custom checkouts in the first place.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -13,8 +13,15 @@ jobs:
     - uses: actions/checkout@v4
       with:
         ref: ${{ github.event.pull_request.head.sha }}
-    - run: npm ci && npm run build
+    - run: npm ci
+    - run: npm run build
```
Comment (zimeg) [inline .github/workflows/main.yml:17]: 👏

**Review by zimeg — APPROVED** [review]
Otherwise, you could full send the change! It looks good to me and I'm thinking running this on `main` will "just work".

**Comment by zimeg** [comment]
Works on `main`! https://github.com/slackapi/slack-github-action/actions/runs/9008574305/job/24750965740

**Pull request merged**